### ISSUE TITLE
Implement Permanent Removal of Resources

### ISSUE THREAD

**Issue by alexanderkiel** [samply/blaze#399]
As Blaze implements versioning of resources, the delete interaction only marks resources as deleted. In some scenarios, for example for GDPR compliance, it might be necessary to really remove a resource from Blaze.

### Other Work

* HAPI has the [$expunge](https://smilecdr.com/docs/fhir_repository/deleting_data.html#expunge) operation
* CRUX hat the [evict](https://opencrux.com/reference/21.06-1.17.1/transactions.html#evict) operation
* Datomic has [Excision](https://docs.datomic.com/on-prem/reference/excision.html)

**Comment by MM-Lehmann** [comment]
Must have (MVP): removal of a single resource by reference (id)
Should have: cascading removal of all resources referencing a particular Patient(-id)
Could have: cascading removal of all resource referencing any given resource

**Comment by MM-Lehmann** [comment]
Is there any other way to completely wipe the server via REST?

**Comment by alexanderkiel** [comment]
> Is there any other way to completely wipe the server via REST?

No, you have to shutdown Blaze, delete the docker volume and restart it.

**Comment by JohannesOehm** [comment]
Are there any update on this? This prevents us currently from using Blaze in our DIC.

**Comment by alexanderkiel** [comment]
@JohannesOehm Would it be sufficient to be able to purge a single resource with all of it's history? That would be "Instance-Level Expunge" in HAPI. Would it be ok if metadata about the transactions that created/updated/deleted the resource will still exist but the resource contents are purged from disk?

**Comment by MM-Lehmann** [comment]
That would be sufficient for us
  

  

  

  
  
  
  
>

**Comment by JohannesOehm** [comment]
It would be better if the resources ID is also deleted, but we can replace the resource IDs, which currently hold the patients pseudonym with some random numbers, so it is also fine for us.
